src/main/java/com/faforever/client/preferences/ui/SettingsController.java

```diff
@@ -310,11 +313,19 @@ public void initialize() {
     autoChannelListView.setSelectionModel(new NoSelectionModel<>());
     autoChannelListView.setFocusTraversable(false);
     autoChannelListView.setItems(preferencesService.getPreferences().getChat().getAutoJoinChannels());
-    autoChannelListView.setCellFactory(param -> uiService.<RemovableListCellController>loadFxml("theme/settings/removable_cell.fxml"));
+    autoChannelListView.setCellFactory(param -> uiService.<RemovableListCellController<String>>loadFxml("theme/settings/removable_cell.fxml"));
     autoChannelListView.getItems().addListener((ListChangeListener<String>) c -> preferencesService.storeInBackground());
     autoChannelListView.managedProperty().bind(autoChannelListView.visibleProperty());
     autoChannelListView.visibleProperty().bind(Bindings.createBooleanBinding(() -> !autoChannelListView.getItems().isEmpty(), autoChannelListView.getItems()));
 
+    mirrorURLsListView.setSelectionModel(new NoSelectionModel<>());
+    mirrorURLsListView.setFocusTraversable(false);
+    mirrorURLsListView.setItems(preferencesService.getPreferences().getMirror().getMirrorURLs());
+    mirrorURLsListView.setCellFactory(param -> uiService.<RemovableListCellController<URI>>loadFxml("theme/settings/removable_cell.fxml"));
+    mirrorURLsListView.getItems().addListener((ListChangeListener<URI>) c -> preferencesService.storeInBackground());
+    mirrorURLsListView.managedProperty().bind(mirrorURLsListView.visibleProperty());
+    mirrorURLsListView.visibleProperty().bind(Bindings.createBooleanBinding(() -> !mirrorURLsListView.getItems().isEmpty(), mirrorURLsListView.getItems()));
```

Comment:
If this is ever updated not through the UI then this needs to be a listener and not a binding

Reply:
Yea idk, I just copy/pasted from the autojoinchannel listview